presto-orc/src/main/java/com/facebook/presto/orc/reader/ColumnReader.java

```diff
@@ -17,63 +17,68 @@
 import com.facebook.presto.orc.QualifyingSet;
 import com.facebook.presto.orc.stream.BooleanInputStream;
 import com.facebook.presto.orc.stream.LongInputStream;
-import com.facebook.presto.spi.block.Block;
 import com.facebook.presto.spi.type.Type;
 
 import java.io.IOException;
 import java.util.Arrays;
 
+import static java.util.Objects.requireNonNull;
+
 abstract class ColumnReader
         implements StreamReader
 {
-    QualifyingSet inputQualifyingSet;
-    QualifyingSet outputQualifyingSet;
-    Block block;
-    int outputChannel = -1;
-    Filter filter;
-    int columnIndex;
-    Type type;
-    int expectNumValues = 10000;
+    protected final int expectNumValues = 10000;
+
+    protected QualifyingSet inputQualifyingSet;
```

Comment:
Should we expose mutable state to child classes? 
Would changing to private and using a method getInputQualifyingSet() be safer?